### PR TITLE
feat: Microvm Cluster controller

### LINKED ISSUE #11: Initial implementation of `MicrovmCluster` controller
**Describe the solution you'd like:**
Implement the initial `MicrovmCluster` controller. There won't be much logic in this initial reconciliation and we should focus on satisfying the CAPI cluster contract.

**Why do you want this feature:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

This will work in progress for the cluster controller

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #11 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: config/crd/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 commonLabels:
-  cluster.x-k8s.io/v1alpha3: v1alpha1
+  cluster.x-k8s.io/v1beta1: v1alpha1
```
Comment (Callisto13) [inline config/crd/kustomization.yaml:2]: ngl i do _not_ understand this label. how does label `v1beta1` equal `v1aplha1`?

File: controllers/microvmcluster_controller.go [hunk 2]
```diff
@@ -38,25 +56,139 @@ type MicrovmClusterReconciler struct {
...
+	if !available {
+		conditions.MarkFalse(clusterScope.MvmCluster, infrav1.LoadBalancerAvailableCondition, infrav1.LoadBalancerNotAvailableReason, clusterv1.ConditionSeverityInfo, "control plane load balancer isn't available")
+
+		return reconcile.Result{RequeueAfter: requeuePeriod}, nil
+	}
+	conditions.MarkTrue(clusterScope.MvmCluster, infrav1.LoadBalancerAvailableCondition)
+
+	if clusterScope.Cluster.Spec.ControlPlaneEndpoint.IsZero() {
+		clusterScope.Cluster.Spec.ControlPlaneEndpoint = *clusterScope.MvmCluster.Spec.ControlPlaneEndpoint.DeepCopy()
+	}
```
Comment (Callisto13) [inline controllers/microvmcluster_controller.go]: think this was the one we agreed can go

**Review by Callisto13 — DISMISSED** [review]

File: config/crd/kustomization.yaml (hunk 1 above)
Comment (richardcase) [reply config/crd/kustomization.yaml:2]: Its a mapping between CAPI API versions and the provider versions. So it this case, CAPI is at v1beta1 but CAPMVM is at v1alpha1

File: controllers/microvmcluster_controller.go (hunk 2 above)
Comment (richardcase) [reply controllers/microvmcluster_controller.go]: Yep, just removed it.

**Comment by Callisto13** [comment]
i'm gonna push a quick fix to:
- fix the linter
- add a test action
- bump to go 1.17

on this branch then we can merge 👍

**Comment by richardcase** [comment]
@Callisto13 - i also updated the go version and lint workflow because it was complaining at me.

**Comment by richardcase** [comment]
Ha, same thinking. Although i haven't added the test action.

**Comment by Callisto13** [comment]
Lol! I can add one now real quick

**Review by Callisto13 — APPROVED** [review]

**Pull request merged**